### PR TITLE
add Jaeger-operator to helm charts

### PR DESCRIPTION
**Pull request by cpanato**
#### What this PR does / why we need it:
Add Jaeger Operator to Helm Charts

#### Which issue this PR fixes
  - fixes # https://github.com/jaegertracing/jaeger-operator/issues/4

#### Checklist
- [x] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md

### CODE REVIEW

File: stable/jaeger-operator/README.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+```console
+$ helm delete my-release
+```
+
+The command removes all the Kubernetes components associated with the chart and deletes the release.
+
+## Configuration
+
+The following table lists the configurable parameters of the kanister-operator chart and their default values.
```
Comment (objectiser) [inline stable/jaeger-operator/README.md]: `jaeger-operator`

File: stable/jaeger-operator/README.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+Parameter | Description | Default
+--- | --- | ---
+`image.repository` | controller container image repository | `kanisterio/controller`
```
Comment (objectiser) [inline stable/jaeger-operator/README.md]: `jaegertracing/jaeger-operator`

File: stable/jaeger-operator/templates/NOTES.txt [hunk 3]
```diff
@@ -0,0 +1,8 @@
+jaeger-operator is installed.
+
+
+Check the jaeger-operator logs
+  export POD=$(kubectl get pods -l app={{ template "jaeger-operator.fullname" . }} --namespace {{ .Release.Namespace }} --output name)
```
Comment (objectiser) [inline stable/jaeger-operator/templates/NOTES.txt]: Should be `export POD=$(kubectl get pods-l app.kubernetes.io/instance={{ .Release.Name }} -lapp.kubernetes.io/name=jaeger-operator --namespace {{ .Release.Namespace }} --output name)`

File: stable/jaeger-operator/README.md (hunk 1 above)
Comment (cpanato) [reply stable/jaeger-operator/README.md]: done. thanks for the catch

File: stable/jaeger-operator/README.md (hunk 2 above)
Comment (cpanato) [reply stable/jaeger-operator/README.md]: done

File: stable/jaeger-operator/templates/NOTES.txt (hunk 3 above)
Comment (cpanato) [reply stable/jaeger-operator/templates/NOTES.txt]: done

File: stable/jaeger-operator/README.md [hunk 4]
```diff
@@ -0,0 +1,63 @@
+# jaeger-operator
+
+[jaeger-operator](https://github.com/jaegertracing/jaeger-operator) is a Kubernetes operator.
+
+## Install
+
+```console
+$ helm install stable/jaeger-operator
```
Comment (jpkrohling) [inline stable/jaeger-operator/README.md:8]: Could this be added to `incubator` for the first couple releases?

Comment (cpanato) [reply stable/jaeger-operator/README.md:8]: afaik we should not use incubator anymore.

@davidkarlsen @mattfarina @unguiculus is correct? or should I move to incubator?

**Comment by jpkrohling** [comment]
Any news about this PR?

File: stable/jaeger-operator/README.md (hunk 4 above)
Comment (davidkarlsen) [reply stable/jaeger-operator/README.md:8]: As far as I have understood the incubator is deprecated and we want to go straight for stable - which makes sense - make it proper 1st time and avoid the hassle of promotion.

File: stable/jaeger-operator/templates/role.yaml [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+  verbs:
+  - "*"
+- apiGroups:
+  - extensions
+  resources:
+  - ingresses
+  verbs:
+  - "*"
+
+---
```
Comment (davidkarlsen) [inline stable/jaeger-operator/templates/role.yaml]: split out the rolebinding into a separate file

**Review by davidkarlsen — CHANGES_REQUESTED** [review]

File: stable/jaeger-operator/templates/role.yaml [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+metadata:
+  name: default-account-{{ include "jaeger-operator.fullname" . }}
+  labels:
+    app.kubernetes.io/name: {{ include "jaeger-operator.name" . }}
+    helm.sh/chart: {{ include "jaeger-operator.chart" . }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
+subjects:
+- kind: ServiceAccount
+  name: default
```
Comment (davidkarlsen) [inline stable/jaeger-operator/templates/role.yaml]: See https://github.com/helm/helm/blob/master/docs/chart_best_practices/rbac.md for how to define the serviceaccount.

**Review by davidkarlsen — CHANGES_REQUESTED** [review]

File: stable/jaeger-operator/templates/role.yaml (hunk 5 above)
Comment (cpanato) [reply stable/jaeger-operator/templates/role.yaml]: done

File: stable/jaeger-operator/README.md (hunk 4 above)
Comment (cpanato) [reply stable/jaeger-operator/README.md:8]: 👍

File: stable/jaeger-operator/templates/role.yaml (hunk 6 above)
Comment (cpanato) [reply stable/jaeger-operator/templates/role.yaml]: thanks for the review and the link. Done. PTAL @davidkarlsen

**Review by davidkarlsen — APPROVED** [review]

**Pull request merged**